src/modules/previewpane/common/PreviewHandler.cs

```diff
@@ -0,0 +1,140 @@
+        public void SetWindow(IntPtr hwnd, ref RECT rect)
+        {
+            this.parentHwnd = hwnd;
+            this.windowBounds = rect.ToRectangle();
+            this.previewControl.SetWindow(hwnd, this.windowBounds);
+        }
+
+        /// <inheritdoc />
+        public void SetRect(ref RECT rect)
+        {
+            this.windowBounds = rect.ToRectangle();
+            this.previewControl.SetRect(this.windowBounds);
+        }
+
+        /// <inheritdoc />
+        public void Unload()
+        {
+            this.previewControl.Unload();
+        }
+
+        /// <inheritdoc />
+        public void SetFocus()
+        {
+            this.previewControl.SetFocus();
+        }
+
+        /// <inheritdoc />
+        public void QueryFocus(out IntPtr phwnd)
+        {
+            this.previewControl.QueryFocus(out IntPtr result);
+            phwnd = result;
+            if (phwnd == IntPtr.Zero)
+            {
+                throw new Win32Exception();
+            }
+        }
+
+        /// <inheritdoc />
+        public uint TranslateAccelerator(ref MSG pmsg)
+        {
+            // Current implementation simply directs all Keystrokes to IPreviewHandlerFrame. This is the recommended approach to handle keystokes for all low-integrity preview handlers.
+            // Source: https://docs.microsoft.com/en-us/windows/win32/shell/building-preview-handlers#ipreviewhandlertranslateaccelerator
+            if (this.frame != null)
+            {
+                return this.frame.TranslateAccelerator(ref pmsg);
+            }
+
+            const uint S_FALSE = 1;
+            return S_FALSE;
+        }
+
+        /// <inheritdoc />
+        public void GetWindow(out IntPtr phwnd)
+        {
+            phwnd = this.previewControl.GetHandle();
+        }
+
+        /// <inheritdoc />
+        public void ContextSensitiveHelp(bool fEnterMode)
+        {
+            // Should always return NotImplementedException. Source: https://docs.microsoft.com/en-us/windows/win32/shell/building-preview-handlers#iolewindowcontextsensitivehelp
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc />
+        public void SetSite(object pUnkSite)
+        {
+            // Implementation logic details: https://docs.microsoft.com/en-us/windows/win32/shell/building-preview-handlers#iobjectwithsitesetsite
+            this.unkSite = pUnkSite;
+            this.frame = this.unkSite as IPreviewHandlerFrame;
+        }
+
+        /// <inheritdoc />
+        public void GetSite(ref Guid riid, out object ppvSite)
+        {
+            ppvSite = this.unkSite;
+        }
+
+        /// <inheritdoc />
+        public void SetBackgroundColor(COLORREF color)
+        {
+            var argbColor = color.Color;
+            this.previewControl.SetBackgroundColor(argbColor);
+        }
+
+        /// <inheritdoc />
+        public void SetFont(ref LOGFONT plf)
+        {
+            var font = Font.FromLogFont(plf);

```

Comment:
Any reason to set the font here?

Reply:
No reason removed the temporary variable.